lib/IDE/REPLCodeCompletion.cpp

```diff
@@ -223,6 +223,10 @@ doCodeCompletion(SourceFile &SF, StringRef EnteredCode, unsigned *BufferID,
   // Now we are done with code completion.  Remove the declarations we
   // temporarily inserted.
   SF.Decls.resize(OriginalDeclCount);
+
+  // Reset the error state because it's only relevant to the code that we just
+  // processed, which now gets thrown away.
+  Ctx.Diags.resetHadAnyError();
```

Comment:
Ah, this isn't correct. There might have been errors *before* we started, right?

Reply:
My guess is that there is never any intentional error state that must be preserved across completion calculations. Of course, there could be some accidental error state from other operations that forgot to reset the error state.

The reason I think this is that I wouldn't expect there to be any "ongoing compilation" happening when this function executes. The only compilations I know about are:
* When you execute a line in the REPL, it compiles that line. Errors from these executions become irrelevant once the execution returns.
* When you ask for a completion, it compiles your code. Errors from these compilations become irrelevant once the completion request returns.

I don't have much experience with any of this code, so I could be totally wrong.

Reply:
I…guess that's probably the case. @dcci, @adrian-prantl, @jimingham, does that sound right for LLDB's REPL? (The integrated REPL is not important.)

Reply:
Ping! Should I wait for an answer to this before merging?

Reply:
I don't know the answer to this. If no one else objects let's give this a try.

Reply:
Seems like there have been no objections yet. I'm going to give it a try!